lib/ignored-paths.js

```diff
@@ -192,34 +192,30 @@ IgnoredPaths.prototype.contains = function(filepath, category) {
 
 /**
  * Returns a list of dir patterns for glob to ignore
- * @returns {string[]} list of glob ignore patterns
+ * @returns {function()} method to check whether a folder should be ignored by glob
  */
-IgnoredPaths.prototype.getIgnoredFoldersGlobPatterns = function() {
-    var dirs = DEFAULT_IGNORE_DIRS;
+IgnoredPaths.prototype.getIgnoredFoldersGlobChecker = function() {
 
-    if (this.options.ignore) {
-
-        /* eslint-disable no-underscore-dangle */
-
-        var patterns = this.ig.custom._rules.filter(function(rule) {
-            return rule.negative;
-        }).map(function(rule) {
-            return rule.origin;
-        });
+    /**
+     * Adds `"*"` at the end of `"node_modules/"`,
+     * so that subtle directories could be re-included by .gitignore patterns
+     * such as `"!node_modules/should_not_ignored"`
+     */
+    var dirs = DEFAULT_IGNORE_DIRS.map(function(dir) {
```

Comment:
It looks like both times that `DEFAULT_IGNORE_DIRS` is used we are adding the `*` to the end, so how about just changing `DEFAULT_IGNORE_DIRS`?  Also, the other usage of `DEFAULT_IGNORE_DIRS` adds a leading slash, so maybe the final result should be:

```
 var DEFAULT_IGNORE_DIRS = [
     "/node_modules/*",
     "/bower_components/*"
 ];
```

I do like the note explaining why the `*` is needed, though!